apps/Welcome/components/yellowCommandLine/cmdLine.c

```diff
@@ -60,8 +79,37 @@ static enum
 }
 InputProcessingMode = INPUT_MODE_MENU;
 
-/// Pointer to the text prompt to display when in text entry mode or password entry mode.
-static const char* PromptPtr;
+/// Buffer containing the error message to be displayed on the screen.
+static char ErrorMsg[256] = "";
+
+/// Timer used to wait for further characters after an ESC character, in case it was caused by
+/// a non-ESC key press, such as the arrow keys.
+static le_timer_Ref_t EscapeTimer;
+
+/// Object in which a single menu entry is stored.
+typedef struct
+{
+    char text[80];   ///< The menu entry's text string (null-terminated).
+    cmdLine_MenuSelectionFunc selectionFunc; ///< Function to be called when entry is selected.
+    void* contextPtr;   ///< Opaque pointer value to be passed to the selectionFunc.
+}
+MenuEntry_t;
+
+/// List of Menu Entries.  Index is menu option number minus 1.
+static MenuEntry_t MenuEntryList[9];
```

Comment:
Can you create a `MAX_MENU_OPTIONS` define?  The literal `9` appears in a few different places in the code.